### PR TITLE
Surangap

### PR DESCRIPTION
**Pull request by surangap**
Fixed plugin autoupdate popup | exposed utils.addEvent via AdapterJS

### CODE REVIEW

File: src/plugin/plugin_manager.js [hunk 1]
```diff
@@ -174,10 +174,23 @@ export function getLatestVersionNumber() {
...
-  if (!pluginObject) return;
-  let current = pluginObject.VERSION;
-  let latest  = getLatestVersionNumber();
-  return utils.versionCompare(latest, current) > 0;
+  return new Promise ((resolve, reject) =>{
+    callWhenPluginReady(() => {
+      if (pluginObject) {
+        var current = pluginObject.VERSION;
+        var latest = getLatestVersionNumber();
+        utils.versionCompare(latest, current) > 0 ? resolve(true) : resolve(false);
```
Comment (johache) [inline src/plugin/plugin_manager.js]: would read nicer as 
resolve(utils.versionCompare(latest, current) > 0);

File: src/plugin/plugin_manager.js [hunk 2]
```diff
@@ -174,10 +174,23 @@ export function getLatestVersionNumber() {
...
+      } else {
+        reject("plugin object is not available");
+      }
+    });
+  });
+}
+
+export function checkForUpdate() {
```
Comment (johache) [inline src/plugin/plugin_manager.js]: I don't think we need to keep this function.
At most, you could have a updateIfNewVersionAvailable() function, which would NOT depend on config.autoUpdate, and would just be 
if (isUpdateAvailable()) {
      updatePlugin();
    }

**Review by johache — CHANGES_REQUESTED** [review]

File: src/plugin/plugin_manager.js [hunk 3]
```diff
@@ -174,10 +174,23 @@ export function getLatestVersionNumber() {
...
+  if (isUpdateAvailable() && config.autoUpdate) {
```
Comment (miniruwan) [inline src/plugin/plugin_manager.js]: Better to change the order of the conditions; Do the config.autoUpdate check first and then do the heavy isUpdateAvailable() check (which involves http GET).

**Review by miniruwan — CHANGES_REQUESTED** [review]

File: src/plugin/plugin_manager.js (hunk 3 above)
Comment (surangap) [reply src/plugin/plugin_manager.js]: function removed since no longer use

**Pull request merged**